doc/command_guidelines.md

```diff
@@ -0,0 +1,63 @@
+e.g. `database list-sku-definitions` instead of `database sku-definitions list`
+- Avoid command subgroups that have no commands. This often happens at the first level of a command branch.
+e.g. `keyvault create` instead of `keyvault vault create` (where `keyvault` only has subgroups and adds unnecessary depth to the tree).
+
+## Standard Command Types
+
+The following are standard names and behavioral descriptions for CRUD commands commonly found within the CLI. These standard command types MUST be followed for consistency with the rest of the CLI.
+
+- `CREATE` - standard command to create a new resource. Usually backed server-side by a PUT request. 'create' commands should be idempotent and should return the resource that was created.
+- `UPDATE` - command to selectively update properties of a resource and preserve existing values. May be backed server-side by either a PUT or PATCH request, but the behavior of the command should always be PATCH-like. All `update` commands should be registered using the `generic_update_command` helper to expose the three generic update properties. `update` commands MAY also allow for create-like behavior (PUTCH) in cases where a dedicated `create` command is deemed unnecessary. `update` commands should return the updated resource.
+- `SET` - command to replace all properties of a resource without preserving existing values, typically backed server-side by a PUT request. This is used when PATCH-like behavior is deemed unnecessary and means that any properties not specifies are reset to their default values. `set` commands are more rare compared to `update` commands. `set` commands should return the updated resource.
+- `SHOW` - command to show the properties of a resource, backed server-side by a GET request.
+- `LIST` - command to list instances of a resource, backed server-side by a GET request. When there are multiple "list-type" commands within an SDK to list resources at different levels (for example, listing resources in a subscription vice in a resource group) the functionality should be exposed by have a single list command with arguments to control the behavior. For example, if `--resource-group` is provided, the command will call `list_by_resource_group`; otherwise, it will call `list_by_subscription`.
+- `DELETE` - command to delete a resource, backed server-side by a DELETE request. Delete commands return nothing on success.
+
+## Non-standard Commands
+
+For commands that don't conform to one of the above-listed standard command patterns, use the following guidance.
+
+- (*) Don't use single word verbs if they could cause confusion with the standard command types. For example, don't use `get` or `new` as these sound functionally the same as `show` and `create` respectively, leading to confusion as to what the expected behavior should be.
+e.g. `new`, `get`
+- (*) Descriptive, hyphenated command names are often a better option than single verbs.
```

Comment:
The use of `(*)` was a confusing to me at first--I thought it was mangled markup. Then I scrolled back to the top and saw.  You might simply call this out in-text:
"Blah blah blah, don't use new *(modules only)*"

Reply:
I do call it out at the top so now that you know what the `(*)` means, does it make sense?
I didn't want to prefix each of those lines with `(command modules only)` as it's kinda verbose.

Reply:
Assuming everyone reads the entire document from the top, and pays attention, then yes.  Many people I think will simply scroll to where ever they think the info they want is, especially if they follow a link to a particular section.  Since it's a help document, being verbose is probably better than being cryptic, but I am fine if you want to leave it as is.